src/background/handlers/fetchAndStoreHistory.js

```diff
@@ -0,0 +1,75 @@
+import pLimit from 'p-limit';
+import { db } from '../initdb.js';
+import { shouldBlockDomain } from '../services/blacklist.js';
+import {
+  storeHistoryItems,
+} from '../services/db.js';
+import { ensureEngineIsReady, classifyURLAndTitle } from '../services/ml.js';
+
+export async function fetchAndStoreHistory(days) {
+  const now = Date.now();
+  const startTime = now - days * 86400e3;
+  const items = await browser.history.search({ text: '', startTime, endTime: now, maxResults: 1e6 });
+
+  const checks = await Promise.all(
+    items.map(async i => ({ item: i, blocked: await shouldBlockDomain(i.url) }))
+  );
+  const validItems = checks.filter(r => !r.blocked).map(r => r.item);
+  if (!validItems.length) return console.log('No valid history items found.');
```

Comment:
you are returning a console.log here that is a problem. you should return a 0 (length) and use a console.debug